.github/workflows/main.yml

```diff
@@ -51,7 +51,7 @@ jobs:
         run: |
           mkdir build
           cd build
-          ../autogen.sh c ${{ matrix.configure }}
+          ../autogen.sh c
```

Comment:
but isn't matrix.configure a good way to support different build configurations?

Reply:
See branch ci-test: https://github.com/kmod-project/kmod/commit/4b2a79776461fa923a8d148f92537b59652075fd and execution: https://github.com/kmod-project/kmod/actions/runs/10687966057

IMO nice thing that the configuration for each run is all inside the matrix definition, and not spread in the tasks.  If you are fine with that I can squash it in your last commit.